graphdatascience/query_runner/gds_arrow_client.py

```diff
@@ -537,11 +553,28 @@ def upload_triplets(
         """
         self._upload_data(graph_name, "triplet", triplet_data, batch_size, progress_callback)
 
+    def __getstate__(self) -> Dict[str, Any]:
+        state = self.__dict__.copy()
+        # Remove the FlightClient as it isn't serializable
+        if "_flight_client" in state:
+            del state["_flight_client"]
+        return state
+
+    def _client(self) -> flight.FlightClient:
+        """
+        Lazy client construction to help pickle this class because a PyArrow
+        FlightClient is not serializable.
```

Comment:
Makes sense, but I am curious; in which situations do you need to serialise this class?

Reply:
I'm not sure at this point but we might have needed it this way on BigQuery Connector

Reply:
yes sure, I will let you know why when I make sure about the need on the connector side

Reply:
In the BigQuery connector, we use Spark jobs to process data, and Spark requires a serialized version of this class to distribute the job across different workers. Since FlightClient is not inherently serializable, we needed this lazy initialization.